src/main/java/com/microsoft/aad/msal4j/Account.java

```diff
@@ -23,4 +24,22 @@
     String environment;
 
     String username;
+
+    String localAccountId;
+
+    Map<String, ?> idTokenClaims;
+
+    Map<String, IAccount> tenantProfiles;
+
+    public String getTenantId() {
+        return localAccountId;
+    }
+
+    public Map<String, ?> getClaims() {
+        return idTokenClaims;
+    }
+
+    public Map<String, IAccount> getTenantProfiles() {
```

Comment:
TenantProfile is not Account, relation between tenantProfile and account is many to one

Reply:
The implementation in the Android library seems to suggest that they are accounts, or at least can be treated like accounts without issue.

If you look at Android's [AccountAdapter.java](https://github.com/AzureAD/microsoft-authentication-library-for-android/blob/dev/msal/src/main/java/com/microsoft/identity/client/AccountAdapter.java), you can see the only place (as far as I can tell) tenant profiles are directly created. There is an adapt() method with is given a list of cache records, for example in MultipleAccountPublicClientApplications.getAccount(), which appears to split a list of ICacheRecord into accounts and tenant profiles solely by comparing the local and home ID values from ICacheRecord.getAccount() , which returns AccountRecord objects. If the AccountRecord's home ID contains the local ID it's either an Account or MultiTenantAccount, and if not it's a tenant profile.

So, if the data in AccountRecords (which seem to be Android's version of Java's AccountCacheEntity) is the only thing that determines what is an Account and what is a TenantProfile, and since we decided not to add all of the extra interfaces/classes that Android and iOS implemented, then isn't a TenantProfile essentially an Account?